mautic/mautic
# Email draft feature

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | Y
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | N
| Automated tests included?              | Y 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

Add `'email_draft_enabled' => true` in local.php for enable email draft functionality.

Users would like to be able to edit a published email and be able to preview changes without them being published.

**Workflow :**
- A user has a published email and wants to make a change to it.
- The user will will create a draft from the email which is a clone of the email HTML content.
- The user will make the changes in the draft email HTML.
- The user may send the draft (webview preview) email to review to other person.
- When the change is approved the user can merge the change  back to the original email or discard draft anytime.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add 'email_draft_enabled' => true in local.php for enable email draft functionality.
3. verify Icon on Email List page to show Draft if draft saved 
4. User must be able to save their edits as a draft from the Edit Email screen using Save draft button
5. Verify Preview Draft URL must reflect draft state if it exists. verify email Preview should work as it is.
6. User must be able to push draft state to published state using Apply Draft button
7. User must be able to discard draft  using discard draft button
